frappe/model/meta.py

```diff
@@ -122,23 +122,23 @@ def process(self):
 	def as_dict(self, no_nulls = False):
 		def serialize(doc):
 			out = {}
-			for key in doc.__dict__:
-				value = doc.__dict__.get(key)
-
+			for key, value in doc.__dict__.items():
 				if isinstance(value, (list, tuple)):
-					if len(value) > 0 and hasattr(value[0], '__dict__'):
-						value = [serialize(d) for d in value]
```

Comment:
This code relied on `__dict__` earlier to serialize documents. I'm using `isinstance` instead. Also added other minor fixes.

Reply:
This doesn't work when child table contains dictionaries (unsaved docs)

But it looks it never worked before also so eh.

Reply:
> This doesn't work when child table contains dictionaries (unsaved docs)

Probably because meta is never _saved_, it's mostly read-only.